CMakeLists.txt

```diff
@@ -99,6 +100,16 @@ MESSAGE (STATUS "Version: ${PROJECT_VERSION_STRING}")
 
 ## Project Configuration
 
+### Clang
```

Comment:
We might want to split the compiler/tooling addition and the changes on celestial constructor

Reply:
Yeah for sure. This is a hangover from a spike my bad.